components/integrations/bots/add_bot/add_bot.jsx

```diff
@@ -0,0 +1,516 @@
+            description: e.target.value,
+        });
+    }
+
+    updatePicture = (e) => {
+        if (e.target.files && e.target.files[0]) {
+            const pictureFile = e.target.files[0];
+            this.previewBlob = URL.createObjectURL(pictureFile);
+
+            var reader = new FileReader();
+            reader.onload = (e2) => {
+                const orientation = this.getExifOrientation(e2.target.result);
+                const orientationStyles = this.getOrientationStyles(orientation);
+
+                this.setState({
+                    image: this.previewBlob,
+                    orientationStyles,
+                });
+            };
+            reader.readAsArrayBuffer(pictureFile);
+            e.target.value = null;
+            this.setState({pictureFile});
+        } else {
+            this.setState({pictureFile: null, image: null});
+        }
+    }
+
+    setDefault = () => {
+        this.setState({pictureFile: 'default', image: BotDefaultIcon});
+    }
+
+    // based on https://stackoverflow.com/questions/7584794/accessing-jpeg-exif-rotation-data-in-javascript-on-the-client-side/32490603#32490603
+    getExifOrientation(data) {
+        var view = new DataView(data);
+
+        if (view.getUint16(0, false) !== 0xFFD8) {
+            return -2;
+        }
+
+        var length = view.byteLength;
+        var offset = 2;
+
+        while (offset < length) {
+            var marker = view.getUint16(offset, false);
+            offset += 2;
+
+            if (marker === 0xFFE1) {
+                if (view.getUint32(offset += 2, false) !== 0x45786966) {
+                    return -1;
+                }
+
+                var little = view.getUint16(offset += 6, false) === 0x4949;
+                offset += view.getUint32(offset + 4, little);
+                var tags = view.getUint16(offset, little);
+                offset += 2;
+
+                for (var i = 0; i < tags; i++) {
+                    if (view.getUint16(offset + (i * 12), little) === 0x0112) {
+                        return view.getUint16(offset + (i * 12) + 8, little);
+                    }
+                }
+            } else if ((marker & 0xFF00) === 0xFF00) {
+                offset += view.getUint16(offset, false);
+            } else {
+                break;
+            }
+        }
+        return -1;
+    }
+
+    getOrientationStyles(orientation) {
+        const {
+            transform,
+            'transform-origin': transformOrigin,
+        } = exif2css(orientation);
+        return {transform, transformOrigin};
+    }
+
+    handleSubmit = async (e) => {
+        e.preventDefault();
+
+        if (this.state.adding) {
+            return;
+        }
+
+        if (!this.state.username) {
+            this.setState({
+                error: (
+                    <FormattedMessage
+                        id='bots.manage.add.invalid_username'
+                        defaultMessage='Please enter a valid username.'
```

Comment:
```suggestion
                        defaultMessage='Usernames have to begin with a lowercase letter and be 3-22 characters long. You can use lowercase letters, numbers, periods, dashes, and underscores.'
```

Reply:
FYI that this is based on new proposed username help texts elsewhere in the UI. Applying here for consistency 

https://projects.invisionapp.com/share/B7R4BUHYZCS#/screens/353633806